src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private Object[] elements;
+    private int size;
+    private final int defaultCapacity = 10;
+
+    public ArrayList() {
+        elements = new Object[defaultCapacity];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds: " + index);
```

Comment:
The exception `ArrayListIndexOutOfBoundsException` is not defined in this code. You should either define this custom exception or use a standard exception like `IndexOutOfBoundsException`.